cfgov/unprocessed/js/organisms/VideoPlayer.js

```diff
@@ -0,0 +1,190 @@
+/* ==========================================================================
+   Base Video Player Class
+   ========================================================================== */
+
+// Required modules.
+import { checkDom, setInitFlag } from '../modules/util/atomic-helpers';
+import youTubeAPI from '../modules/youtube-api';
+import EventObserver from '../modules/util/EventObserver';
+
+const BASE_CLASS = 'o-video-player';
+
+let _dom;
+let _imageDom;
+let _defaultImageURL;
+let _videoId;
+let _playBtnDom;
+let _closeBtnDom;
+let _iframeDom;
+let _player;
+
+let _playBtnClickedHandlerBinded;
+let _closeBtnClickedHandlerBinded;
+let _playVideoBinded;
+let _stopVideoBinded;
+
+/**
+ * VideoPlayer
+ * @class
+ *
+ * @classdesc Initializes a new VideoPlayer organism.
+ *
+ * @param {HTMLNode} element
+ *   The DOM element within which to search for the organism.
+ * @returns {VideoPlayer} An instance.
+ */
+function VideoPlayer( element ) {
+  // Attach event methods.
+  const eventObserver = new EventObserver();
+  this.addEventListener = eventObserver.addEventListener;
+  this.dispatchEvent = eventObserver.dispatchEvent;
+  this.removeEventListener = eventObserver.removeEventListener;
+
+  _dom = checkDom( element, BASE_CLASS );
+  _videoId = _dom.getAttribute( 'data-id' );
+  _imageDom = _dom.querySelector( `.${ BASE_CLASS }_image` );
+  _defaultImageURL = _imageDom.src;
+  _playBtnDom = _dom.querySelector( `.${ BASE_CLASS }_play-btn` );
+  _closeBtnDom = _dom.querySelector( `.${ BASE_CLASS }_close-btn` );
+  _iframeDom = _dom.querySelector( `.${ BASE_CLASS }_iframe` );
+
+  _playBtnClickedHandlerBinded = _playBtnClickedHandler.bind( this );
+  _closeBtnClickedHandlerBinded = _closeBtnClickedHandler.bind( this );
+  _playVideoBinded = playVideo.bind( this );
+  _stopVideoBinded = stopVideo.bind( this );
+
+  // Expose the public methods.
+  this.init = init;
+
+  return this;
+}
+
+/**
+ * @returns {VideoPlayer} An instance.
+ */
+function init() {
+  setInitFlag( _dom );
+
+  _imageLoad();
+
+  youTubeAPI.attachAPIReadyCallback( _videoAPIReady );
+  youTubeAPI.embedVideoScript();
+
+  return this;
+}
+
+function _videoAPIReady() {
+  _player = youTubeAPI.instantiatePlayer( _iframeDom, _videoId );
+  _player.addEventListener( 'onReady', _videoPlayerReadyHandler );
+  _player.addEventListener( 'onStateChange', _videoStateChangeHandler );
+}
+
+/**
+ * Event handler for the when the video is ready.
+ */
+function _videoPlayerReadyHandler() {
+  // Add events.
+  _playBtnDom.addEventListener( 'click', _playBtnClickedHandlerBinded );
+  _closeBtnDom.addEventListener( 'click', _closeBtnClickedHandlerBinded );
+
+  // The video has loaded.
+  _dom.classList.add( `${ BASE_CLASS }__loaded` );
+}
+
+function _videoStateChangeHandler( event ) {
+  if ( event.data === window.YT.PlayerState.ENDED ) {
+    _stopVideoBinded();
+  }
+}
+
+/**
+ * HANDLE THUMBNAIL IMAGE.
```

Comment:
Hmm... there are four functions, it appears, relating to this. Is it worth breaking them out into their own file?

If not, I'd like to see at least some extra blank lines or comment separators or something to make it clearer where this section begins and ends.

Reply:
Added additional code comment section below the four image methods.